e2e/shim_via_docker_test.go

```diff
@@ -14,107 +14,95 @@ import (
-		"docker", "run", "-d",
-		"--network=host",
-		"--runtime", "io.containerd.remoteproc.v1",
-		"--annotation", fmt.Sprintf("remoteproc.name=%s", "other-processor"),
-		"test-image")
-	assert.Error(t, err)
-	assert.Contains(t, stderr, "remote processor other-processor does not exist, available remote processors: a-processor")
-}
-
-func TestDockerKillProcessByPid(t *testing.T) {
-	rootDir := t.TempDir()
-	remoteprocName := "yolo-device"
-	bins, err := repo.BuildBothBins(t.TempDir(), rootDir, limavm.BinBuildEnv)
-	require.NoError(t, err)
-
-	vm, err := limavm.NewWithDocker(rootDir, "../testdata", bins)
-	require.NoError(t, err)
-	defer vm.Cleanup()
-
-	sim := remoteproc.NewSimulator(rootDir).WithName(remoteprocName)
-	if err := sim.Start(); err != nil {
-		t.Fatalf("failed to run simulator: %s", err)
-	}
-	defer func() { _ = sim.Stop() }()
-
-	containerID, stderr, err := vm.RunCommand(
-		"docker", "run", "-d",
-		"--network=host",
-		"--runtime", "io.containerd.remoteproc.v1",
-		"--annotation", fmt.Sprintf("remoteproc.name=%s", remoteprocName),
-		"test-image")
-	require.NoError(t, err, "stderr: %s", stderr)
-	remoteproc.AssertState(t, sim.DeviceDir(), "running")
-
-	stdout, stderr, err := vm.RunCommand("docker", "inspect", "--format={{.State.Pid}}", containerID)
-	require.NoError(t, err, "stderr: %s", stderr)
-	pid, parseErr := strconv.Atoi(strings.TrimSpace(stdout))
-	require.NoError(t, parseErr)
-
-	_, _, err = vm.RunCommand("kill", "-TERM", fmt.Sprintf("%d", pid))
-	require.NoError(t, err)
-	remoteproc.RequireState(t, sim.DeviceDir(), "offline")
-	requireRecentFinishOfDockerContainer(t, vm, containerID)
+	t.Run("basic container lifecycle", func(t *testing.T) {
+		remoteprocName := "yolo-device"
+		sim := remoteproc.NewSimulator(rootpathPrefix).WithName(remoteprocName)
+		if err := sim.Start(); err != nil {
+			t.Fatalf("failed to run simulator: %s", err)
+		}
+		defer func() { _ = sim.Stop() }()
+
+		remoteproc.AssertState(t, sim.DeviceDir(), "offline")
+
+		containerID, stderr, err := vm.RunCommand(
+			"docker", "run", "-d",
+			"--network=host",
+			"--runtime", "io.containerd.remoteproc.v1",
+			"--annotation", fmt.Sprintf("remoteproc.name=%s", remoteprocName),
+			"test-image")
+		require.NoError(t, err, "stderr: %s", stderr)
+		remoteproc.AssertState(t, sim.DeviceDir(), "running")
+
+		_, stderr, err = vm.RunCommand("docker", "stop", containerID)
+		assert.NoError(t, err, "stderr: %s", stderr)
+		remoteproc.AssertState(t, sim.DeviceDir(), "offline")
+		requireRecentFinishOfDockerContainer(t, vm, containerID)
+
+		_, stderr, err = vm.RunCommand("docker", "start", containerID)
+		assert.NoError(t, err, "stderr: %s", stderr)
+		remoteproc.AssertState(t, sim.DeviceDir(), "running")
+
+		_, stderr, err = vm.RunCommand("docker", "stop", containerID)
+		assert.NoError(t, err, "stderr: %s", stderr)
+		remoteproc.AssertState(t, sim.DeviceDir(), "offline")
+		requireRecentFinishOfDockerContainer(t, vm, containerID)
```

Comment:
As mentioned before, do we need to test `stop` twice?

Reply:
AFAIR there was a good reason for this: testing that stop doesn't leave the system in a non-startable state.